core/string/ustring.h

```diff
@@ -83,92 +83,56 @@ class CharProxy {
 	}
 };
 
-/*************************************************************************/
-/*  Char16String                                                         */
-/*************************************************************************/
+template <typename T>
+class CharStringT {
+	friend class String;
```

Comment:
Why `friend class String`?

Reply:
I don't remember why this was here, but it wasn't needed after the refactor